teps/0087-custom-tasks-graduation.md

```diff
@@ -0,0 +1,437 @@
+we will have nightly releases of each `Custom Task` to catch any failures. Each *stable* `Custom Task` will be used in 
+dogfooding, and will have detailed documentation and multiple examples.
+
+We can explore a bundling releases of Tekton Pipelines and its latest compatible *stable* `Custom Tasks`, such that 
+users can install Tekton Pipelines and `Custom Tasks` together.
+
+#### Graduation Requirements
+
+We can consider graduating a given *experimental* `Custom Task` to *stable* if it meets the following requirements:
+
+1. It is thoroughly tested
+2. It has nightly releases
+3. It is used in dogfooding
+4. It provides several examples or samples 
+5. It has detailed documentation for installation and usage
+6. It has received feedback from the community (solicited in working group, mailing list or other channels)
+
+#### Graduation Process
+
+The graduation process from *experimental* to *stable* for a given `Custom Task` would be:
+
+1. Its Owners validate that the *experimental* `Custom Task` meets the above requirements
+2. Its Owners open a Tekton Enhancement Proposal (TEP) for the functionality provided by the `Custom Task` with its 
+motivation (goals, use cases and requirements), proposal (with design details and evaluation), and alternatives
+3. Its Owners nominate the `Custom Task` for graduation from *experimental* to *stable*
+4. When the Tekton Pipelines Owners have approved the TEP as implementable, the `Custom Task`'s Owners can migrate it to
+the *custom-tasks* repository
+
+### Integrated
+
+When a given *stable* `Custom Task` is actively used and needed for common Continuous Delivery use cases, we can 
+consider making it *integrated* - meaning that its functionality is natively supported in the Tekton Pipelines API. 
+Not all *stable* `Custom Tasks` have to be *integrated*, a *stable* `Custom Task` that improves user experience but
+is not really necessary in Tekton Pipelines API can stay as a *stable* `Custom Task` in the long term. 
+
+#### Graduation Requirements
+
+We can consider graduating a given *stable* `Custom Task` to *integrated* if it meets the following requirements:
+
+1. Its functionality is critical to common Continuous Delivery use cases
+2. It's actively used by the community and in dogfooding
+
+#### Graduation Process
+
+The graduation process from *stable* to *integrated* for a given `Custom Task` would be:
+
+1. Its Owners validate that the *stable* `Custom Task` meets the above requirements
+2. Its Owners update the `Custom Task`'s TEP with:
+   1. a nomination for graduation from *stable* to *integrated*
+   2. a discussion of how its been useful and why it's necessary to natively support the functionality in the 
+   Tekton Pipelines API
+   3. proposal and alternatives for its syntax in the Tekton Pipelines API
+3. When the Tekton Pipelines Owners have approved the TEP as implementable, the `Custom Task`'s Owners will add its 
```

Comment:
good catch -- it could be clearer to create a new tep for integrating the custom task and have a reference to the other tep, so each graduation will have its own tep, what do you think?

Reply:
also, according to the tep process, once a tep is marked as `implemented` it should be *read-only*

https://github.com/tektoncd/community/blob/main/teps/0001-tekton-enhancement-proposal-process.md#tep-workflow

Reply:
so updated it to create a new tep, then we mark the old tep as superseded-by the new tep